zerver/views/realm.py

```diff
@@ -637,6 +637,24 @@ def update_realm_user_settings_defaults(
     if notification_sound is not None or email_notifications_batching_period_seconds is not None:
         check_settings_values(notification_sound, email_notifications_batching_period_seconds)
 
+    if (
+        dense_mode
+        and web_font_size_px is not None
+        and web_font_size_px != RealmUserDefault.WEB_FONT_SIZE_PX_LEGACY
+    ):
+        raise JsonableError(
+            _("Incompatible values for 'dense_mode' and 'web_font_size_px' settings.")
+        )
+
+    if (
+        dense_mode
+        and web_line_height_percent is not None
+        and web_line_height_percent != RealmUserDefault.WEB_LINE_HEIGHT_PERCENT_LEGACY
```

Comment:
Not sure whether we want to also not allow combinations like setting `dense_mode=False` and the font size and line height settings are set to legacy values.

Reply:
I think that's probably fine to permit, I see no problems with it.